sugoi-api-jms-store-provider/src/main/java/fr/insee/sugoi/jms/JmsWriterStore.java

```diff
@@ -51,215 +59,315 @@ public JmsWriterStore(
   }
 
   @Override
-  public void changePassword(
-      User user, String oldPassword, String newPassword, PasswordChangeRequest pcr) {
+  public ProviderResponse changePassword(
+      String userId,
+      String oldPassword,
+      String newPassword,
+      PasswordChangeRequest pcr,
+      ProviderRequest providerRequest) {
     Map<String, Object> params = new HashMap<>();
-    params.put(JmsAtttributes.USER, user);
+    params.put(JmsAtttributes.USER_ID, userId);
     params.put(JmsAtttributes.NEW_PASSWORD, newPassword);
     params.put(JmsAtttributes.OLD_PASSWORD, oldPassword);
     params.put(JmsAtttributes.REALM, realm.getName());
     params.put(JmsAtttributes.USER_STORAGE, userStorage.getName());
     params.put(JmsAtttributes.PASSWORD_CHANGE_REQUEST, pcr);
-    jmsWriter.writeRequestInQueue(queueRequestName, Method.CHANGE_PASSWORD, params);
+    return CheckAndSend(Method.CHANGE_PASSWORD, params, userId, providerRequest);
   }
 
   @Override
-  public void addAppManagedAttribute(String userId, String attributeKey, String attribute) {
+  public ProviderResponse addAppManagedAttribute(
+      String userId, String attributeKey, String attribute, ProviderRequest providerRequest) {
     Map<String, Object> params = new HashMap<>();
     params.put(JmsAtttributes.USER_ID, userId);
     params.put(JmsAtttributes.ATTRIBUTE_KEY, attributeKey);
     params.put(JmsAtttributes.ATTRIBUTE_VALUE, attribute);
-    jmsWriter.writeRequestInQueue(queueRequestName, Method.ADD_APP_MANAGED_ATTRIBUTE, params);
+    return CheckAndSend(Method.ADD_APP_MANAGED_ATTRIBUTE, params, userId, providerRequest);
   }
 
   @Override
-  public void deleteAppManagedAttribute(String userId, String attributeKey, String attribute) {
+  public ProviderResponse deleteAppManagedAttribute(
+      String userId, String attributeKey, String attribute, ProviderRequest providerRequest) {
     Map<String, Object> params = new HashMap<>();
     params.put(JmsAtttributes.USER_ID, userId);
     params.put(JmsAtttributes.ATTRIBUTE_KEY, attributeKey);
     params.put(JmsAtttributes.ATTRIBUTE_VALUE, attribute);
     params.put(JmsAtttributes.REALM, realm.getName());
     params.put(JmsAtttributes.USER_STORAGE, userStorage.getName());
-    jmsWriter.writeRequestInQueue(queueRequestName, Method.DELETE_APP_MANAGED_ATTRIBUTE, params);
+    return CheckAndSend(Method.DELETE_APP_MANAGED_ATTRIBUTE, params, userId, providerRequest);
+  }
+
+  private ProviderResponse CheckAndSend(
+      String method, Map<String, Object> params, String entityId, ProviderRequest providerRequest) {
+    ProviderResponse response = new ProviderResponse();
+
+    // Perhaps use 2 jms template, one with a "reasonable" timeout for synchronous
+    // requeste,
+    // one without any timout, for ckeing status
+    // => the timeout is defined only at jmsTemplate level, not at the receive
+    // method
+
+    // Check status
+    if (providerRequest.getTransactionId() != null
```

Comment:
if a client wants to know the status of its request they have to pass the request again with the id they received in the first response ?

Reply:
yes !! the only problem can be if one user ask the status of a request twice, the first time he will have the good status, but the second time not. because the response message will have been enqueued by the first reading on the broker response queue